### PR TITLE
Add switch bar

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add ha-bar-switch (vertical and horizontal).

This switch will be used in future more info controls
Size, colors and icons are customizable.

### Video demo

https://user-images.githubusercontent.com/5878303/203269517-bf5e575f-dc37-47ed-94ad-cb377de95cf2.mp4

### Interactive demo
https://deploy-preview-14429--home-assistant-gallery.netlify.app/#components/ha-bar-switch

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: gallery/src/pages/components/ha-bar-switch.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import "@material/mwc-switch";
```
Comment (bramkragten) [inline gallery/src/pages/components/ha-bar-switch.ts:1]:
```suggestion
```

**Review by bramkragten — DISMISSED** [review]

**Comment by bramkragten** [comment]
Should we allow dragging the handle?

**Review by bramkragten — APPROVED** [review]

**Comment by piitaya** [comment]
> Should we allow dragging the handle?

For now, only toggle but it could be a good improvement 🙂

**Comment by matthiasdebaat** [comment]
> I am a little bit surprised about this, because until now Home Assistant uses the Material Design language and I think @matthiasdebaat said somewhere, he wants to migrate to material 3.
> 
> But then this new component is totally different from how material design looks like.
> 
> Has the desire to follow material design changed and home assistant now plans to ship its own in house design?

That's correct. These components aren't in the M3 library, some are software specific. In this case we made these bars, based on the Material design language. Can you elaborate why find it totally different?

**Comment by piitaya** [comment]
These component will be mainly used in the new more info dialogs as presented during the state of open home : https://www.figma.com/proto/ZPRTON7huEj9kyXzPNxNgK/Future-concept?node-id=263%3A6636&comments-enabled=1

CleanShot 2022-11-30 at 15 32 45@2x

**Comment by LasseRosenow** [comment]
> These component will be mainly used in the new more info dialogs as presented during the state of open home : https://www.figma.com/proto/ZPRTON7huEj9kyXzPNxNgK/Future-concept?node-id=263%3A6636&comments-enabled=1
> 
> CleanShot 2022-11-30 at 15 32 45@2x

Oh I see that kinda makes sense. I guess the material library does not have so large components that really fit. Btw. This design of how home assistant is going to look like is really amazing :o I literally cannot wait to see this new design in action 😍😍😍😍😍

> That's correct. These components aren't in the M3 library, some are software specific. In this case we made these bars, based on the Material design language. Can you elaborate why find it totally different?

I am sorry I tried to wrap my head around it, but I think I lack the expertise to give a good answer to this. Especially after the use case for this component now is clear to me I don't know how to make them look more material. Maybe one thing is colors? They seem a bit not so material color like, but this is not really a problem of the component itself haha.

I guess the design in general feels a bit not so material also because there are material widgets that have the same functionality in theory like the `Material Switch` and `Material Slider`, but just look very different.
Like the slider approach of material is to have a line with a dot and you would not have these kind of handle lines like in your version. And the switch is usually totally round. 

But I agree that those material widgets don't fit for this use case :)

**Pull request merged**